File: client/src/types/product-variant.ts
export interface CreateProductVariantRequest {
    productId: string
    name: string
    size: string
    scent: string
    stock: number
    price: number
  }
  
  export interface UpdateProductVariantRequest {
    name: string
    size: string
    scent: string
    stock: number
    price: number
  }
  
  export interface ProductVariant {
    id: string
    name: string
    size: string
    scent: string
    stock: number
    price: number
    productId: string
  }

File: server/src/main/java/com/fragrance/raumania/model/product/ProductVariant.java
package com.fragrance.raumania.model.product;

import com.fragrance.raumania.model.cart.CartItem;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;

import java.sql.Types;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Entity
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductVariant {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @JdbcTypeCode(Types.VARCHAR)
    private UUID id;

    private String name;

    private String size;

    private String scent;

    private Integer stock;

    private Double price;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private Product product;

    @Builder.Default
    @OneToMany(mappedBy = "productVariant", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<CartItem> cartItems = new HashSet<>();
}
